substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -261,19 +261,23 @@ impl ChainSpecBuilder {
 						.map_err(|e| format!("Conversion to json failed: {e}"))?;
 
 				// We want to extract only raw genesis ("genesis::raw" key), and apply it as a patch
-				// for the original json file. However, the file also contains original plain
-				// genesis ("genesis::runtimeGenesis") so set it to null so the patch will erase it.
+				// for the original json file.
 				genesis_json.as_object_mut().map(|map| {
 					map.retain(|key, _| key == "genesis");
+				});
+
+				let mut org_chain_spec_json = extract_chain_spec_json(input_chain_spec.as_path())?;
+
+				// The original plain genesis ("genesis::runtimeGenesis") is no longer needed, so
+				// just remove it:
+				org_chain_spec_json.as_object_mut().map(|map| {
 					map.get_mut("genesis").map(|genesis| {
-						genesis.as_object_mut().map(|genesis_map| {
-							genesis_map
-								.insert("runtimeGenesis".to_string(), serde_json::Value::Null);
-						});
+						genesis
+							.as_object_mut()
+							.map(|genesis_map| genesis_map.remove("runtimeGenesis"));
```

Comment:
nit:  Would something like this work? arguably looks cleaner

```rust
org_chain_spec_json
   .get_mut("genesis")
   .and_then(|inner| inner.as_object_mut())
   .map(|inner| inner.remove("runtimeGenesis"));
```

Reply:
c95cfa1

Reply:
yeah, even better: 62a780f